### DISCUSSION TITLE
Cypress error: certificate has expired in existing project - is-installed-globally - fs-extra lib fails

### DISCUSSION THREAD

**Discussion by Ragunath** [cypress-io/cypress#29203] [Cypress Cloud]
Hi Team,

As we are using old versions of cypress and trying to upgrade it.

Angular 11:
Cypress: 4.12.1
Node 14.20.0 in local and 16.20.0 in Azure pipeline.

We are getting below certificate issues in the pipeline, but not locally.

npm ERR! code 1
npm ERR! path C:\a\9\s\node_modules\cypress
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node index.js --exec install
npm ERR! Installing Cypress (version: 4.2.0)
npm ERR!
npm ERR! 25l[09:10:23] Downloading Cypress [started]
npm ERR! [09:10:23] Downloading Cypress [failed]
npm ERR! 25hThe Cypress App could not be downloaded.
npm ERR!
npm ERR! Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration
npm ERR!
npm ERR! Otherwise, please check network connectivity and try again:
npm ERR!
npm ERR! ----------
npm ERR!
npm ERR! URL: https://download.cypress.io/desktop/4.2.0?platform=win32&arch=x64
npm ERR! Error: certificate has expired
npm ERR!
npm ERR! ----------
npm ERR!
npm ERR! Platform: win32 (10.0.20348)
npm ERR! Cypress Version: 4.2.0
npm ERR! 25h

Also, I tried to upgrade the Cypress version from 4.12.1 to 5/6/7/8.. latest. It throws different issue as

cypress@8.7.0 postinstall C:\Users\node_modules\cypress
node index.js --exec install

C:\Users\node_modules\cypress\node_modules\is-installed-globally\index.js:12
} catch {
^

SyntaxError: Unexpected token {
at createScript (vm.js:80:10)
at Object.runInThisContext (vm.js:139:10)
at Module._compile (module.js:617:28)
at Object.Module._extensions..js (module.js:664:10)
at Module.load (module.js:566:32)
at tryModuleLoad (module.js:506:12)
at Function.Module._load (module.js:498:3)
at Module.require (module.js:597:17)
at require (internal/module.js:11:18)
at Object. (C:\Users\node_modules\cypress\lib\util.js:43:29)
at Module._compile (module.js:653:30)
at Object.Module._extensions..js (module.js:664:10)
at Module.load (module.js:566:32)
at tryModuleLoad (module.js:506:12)
at Function.Module._load (module.js:498:3)
at Module.require (module.js:597:17)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cypress@8.7.0 postinstall: node index.js --exec install
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the cypress@8.7.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Please suggest any solution

I can successfully install in new Angular 11 project. It Occurs only in existing project

I keep changing the version of Cypress ,node version and npm. Now I get this issue..

Cypress: 5.x
node:16.20.1
npm: 8.19.1

verbose stack Error: command failed
7700 verbose stack at ChildProcess. (C:\Users\Tester\AppData\Roaming\npm\node_modules\npm\node_modules@npmcli\promise-spawn\lib\index.js:63:27)
7700 verbose stack at ChildProcess.emit (node:events:527:28)
7700 verbose stack at maybeClose (node:internal/child_process:1090:16)
7700 verbose stack at ChildProcess._handle.onexit (node:internal/child_process:302:5)
7701 verbose pkgid cypress@5.6.0
7702 verbose cwd C:\Users\Tester\source\repos\MyProject
7703 verbose Windows_NT 10.0.19045
7704 verbose node v18.0.0
7705 verbose npm v8.19.4
7706 error code 1
7707 error path C:\Users\Tester\source\repos\MyProject\node_modules\cypress
7708 error command failed
7709 error command C:\Windows\system32\cmd.exe /d /s /c node index.js --exec install
7710 error C:\Users\Tester\source\repos\MyProject\node_modules\cypress\node_modules\fs-extra\lib\mkdirs\make-dir.js:85
7710 error } catch {
7710 error ^
7710 error
7710 error SyntaxError: Unexpected token {
7710 error at createScript (vm.js:80:10)
7710 error at Object.runInThisContext (vm.js:139:10)
7710 error at Module._compile (module.js:617:28)
7710 error at Object.Module._extensions..js (module.js:664:10)
7710 error at Module.load (module.js:566:32)
7710 error at tryModuleLoad (module.js:506:12)
7710 error at Function.Module._load (module.js:498:3)
7710 error at Module.require (module.js:597:17)
7710 error at require (internal/module.js:11:18)
7710 error at Object. (C:\Users\Tester\source\repos\MyProject\node_modules\cypress\node_modules\fs-extra\lib\mkdirs\index.js:3:44)
7710 error at Module._compile (module.js:653:30)
7710 error at Object.Module._extensions..js (module.js:664:10)
7710 error at Module.load (module.js:566:32)
7710 error at tryModuleLoad (module.js:506:12)
7710 error at Function.Module._load (module.js:498:3)
7710 error at Module.require (module.js:597:17)
7711 verbose exit 1
7712 timing npm Completed in 105765ms
7713 verbose unfinished npm timer reify 1711542820193
7714 verbose unfinished npm timer reify:build 1711542902190
7715 verbose unfinished npm timer build 1711542902194
7716 verbose unfinished npm timer build:deps 1711542902195
7717 verbose unfinished npm timer build:run:postinstall 1711542902260
7718 verbose unfinished npm timer build:run:postinstall:node_modules/cypress 1711542902260
7719 verbose code 1

Any solutions please

**Comment by anyuan95** [comment]
```
node_modules/.pnpm/cypress@8.7.0/node_modules/cypress: Running postinstall script, failed in 1.7s
.../cypress@8.7.0/node_modules/cypress postinstall$ node index.js --exec install
│ Installing Cypress (version: 8.7.0)
│ [STARTED] Task without title.
│ [FAILED] The Cypress App could not be downloaded.
│ [FAILED] 
│ [FAILED] Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configurati…
│ [FAILED] 
│ [FAILED] Otherwise, please check network connectivity and try again:
│ [FAILED] 
│ [FAILED] ----------
│ [FAILED] 
│ [FAILED] URL: https://download.cypress.io/desktop/8.7.0?platform=linux&arch=x64
│ [FAILED] Error: certificate has expired
│ [FAILED] 
│ [FAILED] ----------
│ [FAILED] 
│ [FAILED] Platform: linux (CentOS - 7.1.1503)
│ [FAILED] Cypress Version: 8.7.0
│ The Cypress App could not be downloaded.
│ Does your workplace require a proxy to be used to access the Internet? If so, you must configure the HTTP_PROXY environment variable before downloading Cypress. Read more: https://on.cypress.io/proxy-configuration
│ Otherwise, please check network connectivity and try again:
│ ----------
│ URL: https://download.cypress.io/desktop/8.7.0?platform=linux&arch=x64
│ Error: certificate has expired
│ ----------
│ Platform: linux (CentOS - 7.1.1503)
│ Cypress Version: 8.7.0
```

i got the same issue on linux :(

**Reply by anyuan95** [reply]
```
yum update ca-certificates
```
The problem disappears after i update the 'ca-certificates' 🤔🤔
i wonder why...

**Reply by Ragunath** [reply]
This solution doesn't work for Windows :(
